lxd/instance/drivers/driver_common.go

```diff
@@ -850,6 +863,64 @@ func (d *common) snapshotCommon(inst instance.Instance, name string, expiry *tim
 		}
 	})
 
+	// Snapshot attached disk volumes.
+	if len(attachedVolumes) > 0 {
+		// Snapshot attached custom volumes.
+		storageCache := storagePools.NewStorageCache(pool) // Create storage cache for pool lookups.
+		attachedVolumeUUIDs := make(map[string]string)
+		instanceProject := inst.Project()
+		instanceType := inst.Type()
+		instanceName := inst.Name()
+		for _, volume := range attachedVolumes {
+			d.logger.Debug("Creating attached volume snapshot", logger.Ctx{"pool": volume.Pool, "volume": volume.Name, "project": volume.Project})
+
+			// Use shutdown context as we don't have access to the request context.
+			snapshotName, err := storagePools.VolumeDetermineNextSnapshotName(d.state.ShutdownCtx, d.state, volume.Pool, volume.Name, volume.Config)
+			if err != nil {
+				return err
+			}
+
+			// Attached volume snapshot description.
+			description := "Created alongside " + instanceType.String() + " " + instanceName + "/" + snapshotName + " in project " + instanceProject.Name
+
+			// Storage cache lookup.
+			pool, err := storageCache.GetPool(volume.Pool)
+			if err != nil {
+				return err
+			}
+
+			expiry := snap.ExpiryDate() // Attached volume snapshots inherit the expiry date of the instance snapshot.
+			snapshotUUID, err := pool.CreateCustomVolumeSnapshot(volume.Project, volume.Name, snapshotName, description, &expiry, d.op)
+			if err != nil {
+				return err
```

Comment:
lets wrap this error and be sure to include volume name and pool name as it may be different than the instance's pool name